lib/ts/recipe/session/api/implementation.ts

```diff
@@ -14,6 +15,27 @@ export default function getAPIInterface(): APIInterface {
             options: APIOptions;
             userContext: UserContext;
         }): Promise<SessionContainerInterface> {
+            // If a request has multiple session cookies and 'olderCookieDomain' is
+            // unset, we can't identify the correct cookie for refreshing the session.
+            // Using the wrong cookie can cause an infinite refresh loop. To avoid this,
+            // we throw a 500 error asking the user to set 'olderCookieDomain'.
+            if (
+                (hasMultipleCookiesForTokenType(options.req, "access") ||
+                    hasMultipleCookiesForTokenType(options.req, "refresh")) &&
+                options.config.olderCookieDomain === undefined
+            ) {
+                throw new Error(
+                    `The request contains multiple session cookies. This may happen if you've changed the 'cookieDomain' setting in your configuration. To clear tokens from the previous domain, set 'olderCookieDomain' in your config.`
+                );
+            }
```

Comment:
What do you think about moving this into `clearSessionCookiesFromOlderCookieDomain`? My only reason for it is that then we can't forget this check in the future if `clearSessionCookiesFromOlderCookieDomain` is called from multiple places.

Reply:
Yeah that makes more sense. I have updated the PR.